client/src/components/DashboardLayout.tsx

```diff
@@ -0,0 +1,217 @@
+import { HamburgerMenuIcon, SunIcon } from '@radix-ui/react-icons';
+import { useTheme } from 'next-themes';
+import Image from 'next/image';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { useEffect, useState } from 'react';
+import { Box } from '../ui/Box';
+import { Button } from '../ui/Button';
+import { Container } from '../ui/Container';
+import { Dialog, DialogContent, DialogTrigger } from '../ui/Dialog';
+import { Flex } from '../ui/Flex';
+import { IconButton } from '../ui/IconButton';
+
+interface NavItemProps {
+  href: string;
+  children: React.ReactNode;
+}
+
+const NavItem = ({ href, children }: NavItemProps) => {
+  const router = useRouter();
+  return (
+    <Link href={href} passHref>
+      <Box
+        css={{
+          py: '$3',
+          px: '$3',
+          br: '$2',
+          backgroundColor: router.pathname === href ? '$gray5' : 'transparent',
+
+          '&:hover': {
+            backgroundColor: '$gray4',
+          },
+
+          '&:active': {
+            backgroundColor: '$gray5',
+          },
+        }}
+        target={href.includes('https') ? '_blank' : undefined}
+        rel="noreferrer"
+        as="a"
+      >
+        {children}
+      </Box>
+    </Link>
+  );
+};
+
+interface LayoutProps {
+  children?: React.ReactNode;
+}
+
+export const DashboardLayout = ({ children }: LayoutProps) => {
+  const { theme, setTheme } = useTheme();
+  const [mounted, setMounted] = useState(false);
+
+  useEffect(() => setMounted(true), []);
+
+  return (
+    <>
+      <Dialog>
+        <DialogTrigger asChild>
+          <IconButton
+            css={{
+              alignItems: 'center',
+              justifyContent: 'space-between',
+
+              '@lg': {
+                display: 'none',
+              },
+            }}
+          >
+            <HamburgerMenuIcon />
+          </IconButton>
+        </DialogTrigger>
+        <DialogContent>
+          <Flex
+            css={{ flex: 1 }}
+            as={'nav'}
+            direction={'column'}
+            justify={'between'}
+            align={'start'}
+          >
+            <Box>
+              {mounted === true ? (
```

Comment:
to remove the need for the mounted hook (which cause a rerender when mounted), I think it's better to use the solution recommended by next-themes https://github.com/pacocoursey/next-themes#avoid-hydration-mismatch (scroll a bit to "For example, with [next/image](https://nextjs.org/docs/basic-features/image-optimization) you can use an empty image until the theme is resolved.").

Reply:
ahh yes this is a much better solution, thanks! I'm finding however, that when adding an empty image as the default like in the example, it's not seeming to resolve to light or dark and stays empty. Any idea why this could be?

Reply:
Do you see anything special in the console that could help to debug? From my testing, I faced this issue when I migrated part of the images on the page. It was working when all the images displayed were migrated to this new approach :)

Reply:
Yeah I seem to be getting this error: 
![Screenshot 2022-02-14 at 16 20 15](https://user-images.githubusercontent.com/65421744/153903714-9c0defa4-5746-4431-b57b-874ff28bfd77.png)

I believe all images on the page are now using the new approach, but seem to be still having the same issue 🤔

Reply:
Could you push the changes so I can have a look?

Reply:
Just pushed changes :)